### PR TITLE
Add PyPyDispatcher

### PR DESCRIPTION
**Pull request by Gallaecio**
Based on the PyDispatcher receipe.

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by Gallaecio** [comment]
> GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.

@dangra @wRAR @rmax

**Comment by wRAR** [comment]
Sure.

File: recipes/pypydispatcher/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  noarch: python
+  number: 0
+  script: python -m pip install --no-deps --ignore-installed .
```
Comment (chrisburr) [inline recipes/pypydispatcher/meta.yaml:17]:
```suggestion
  script: python -m pip install . -vv
```

File: recipes/pypydispatcher/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
+{% set name = "PyPyDispatcher" %}
+{% set version = "2.1.2" %}
+{% set sha256 = "b6bec5dfcff9d2535bca2b23c80eae367b1ac250a645106948d315fcfa9130f2" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
```
Comment (chrisburr) [inline recipes/pypydispatcher/meta.yaml:10]:
```suggestion
```

File: recipes/pypydispatcher/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
```
Comment (chrisburr) [inline recipes/pypydispatcher/meta.yaml:12]:
```suggestion
  sha256: b6bec5dfcff9d2535bca2b23c80eae367b1ac250a645106948d315fcfa9130f2
```

File: recipes/pypydispatcher/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  license: BSD-3-Clause-Clear
+  license_family: BSD
+  license_file: license.txt
+  summary: 'Multi-producer-multi-consumer signal dispatching mechanism, with PyPy support'
+
+  description: |
+    Multi-producer-multi-consumer signal dispatching mechanism, with PyPy support
+
+extra:
+  recipe-maintainers:
```
Comment (chrisburr) [inline recipes/pypydispatcher/meta.yaml:42]: @dangra @rmax Are you happy to be listed as recipe maintainers here?

**Comment by rmax** [comment]
Nice

File: recipes/pypydispatcher/meta.yaml (hunk 4 above)
Comment (Gallaecio) [reply recipes/pypydispatcher/meta.yaml:40]: @chrisburr I’ve just removed Daniel from the list of recipe maintainers.

@rmax Was https://github.com/conda-forge/staged-recipes/pull/14064#issuecomment-785698563 an OK to be a maintainer? :slightly_smiling_face:

Comment (rmax) [reply recipes/pypydispatcher/meta.yaml:40]: Yes. I'm OK to be listed as maintainer.

**Pull request merged**